### DISCUSSION TITLE
Acoforms Stamper

### DISCUSSION THREAD

**Discussion by topcat30** [UglyToad/PdfPig#427] [Q&A] [answered]
Hoping soneone can help me with Filling an Acroform PDF. Below you will see how i am reading meta data from xml and matching the acroform fileds and filling them with data from the xml. The current program is using itext. How can do this using pigpig?

using System;
using System.Collections;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using iTextSharp;
using iTextSharp.text;
using iTextSharp.text.pdf;
using iTextSharp.text.xml;
using System.IO;
using System.Collections.Generic;

namespace PDFTools
{
    class FillPDF
    {
        public string Fill(string vAttachment, Dictionary<string, string> metadata)
        {
            string outPDF = "";

            byte[] PDFstr = Convert.FromBase64String(vAttachment);
            MemoryStream mstream = new MemoryStream(PDFstr);

            // open the file
            //PdfReader pdf_file = new PdfReader(@"C:\Users\tcitrine.AU-BOTTOMLINE\Desktop\Test.pdf");
            PdfReader pdfReader = new PdfReader(mstream.ToArray());

            MemoryStream msOutPDF = new MemoryStream();

            PdfStamper pdfStamper = new PdfStamper(pdfReader, msOutPDF, '\0', true);

            AcroFields pdfFormFields = pdfStamper.AcroFields;

            foreach (KeyValuePair<string, AcroFields.Item> pdfField in pdfReader.AcroFields.Fields)
            {
                string MetaValue;
                string CurrentField = pdfField.Key.ToString();
                if (metadata.TryGetValue(CurrentField, out MetaValue))
                {
                    pdfFormFields.SetField(CurrentField, MetaValue);

                }
            }

            pdfStamper.FormFlattening = false;
            // close the pdf  
            pdfStamper.Close();

            // outPDF = Encoding.ASCII.GetString(msOutPDF.ToArray());
            //msOutPDF.Close();

            pdfReader.Close();
            mstream.Dispose();
            string tempPDF = Convert.ToBase64String(msOutPDF.ToArray());
            msOutPDF.Dispose();

            byte[] PDFstr2 = Convert.FromBase64String(tempPDF);
            MemoryStream mstream2 = new MemoryStream(PDFstr2);
            PdfReader pdfReader2 = new PdfReader(mstream2.ToArray());
            MemoryStream msOutPDF2 = new MemoryStream();
            PdfStamper pdfStamper2 = new PdfStamper(pdfReader2, msOutPDF2, '\0', false);
            pdfStamper2.FormFlattening = true;
            pdfStamper2.Close();
            outPDF = Convert.ToBase64String(msOutPDF2.ToArray());
            msOutPDF2.Dispose();
            pdfReader.Dispose();
            pdfReader2.Dispose();

            return outPDF;
        }

        

    }
}

**Comment by EliotJones** [comment, marked as answer]
Hi @topcat30 PdfPig doesn't currently support editing AcroForms unfortunately.

**Reply by topcat30** [reply]
Thanks Elliot

**Comment by radrex** [comment]
Hello @EliotJones are there any plans to implement AcroForms editing functionality in the future? Thank you.
